File: 通用代码/命令模式(Reciver封装)(C++)/Command.cpp
#include "Command.h"

Command::Command()
{
    cout << "Command..." << endl;
}

Command::Command(Receiver *pReceiver)
{
    cout << "Command..." << endl;
    this->_pReciever = pReceiver;
}

Command::~Command()
{
    cout << "~Command..." << endl;
    delete this->_pReciever;
    this->_pReciever = NULL;
}

File: 模式混编/命令模式+责任链模式>Unix命令/命令模式部分的设计/Command.java
public abstract class Command{
    //定义一个子类的全局共享变量
    protected final Receiver vo;
    //实现必须定义一个接收者
    public Command(Receiver _vo){
        this.vo = _vo;
    }
    //每个命令类都必须有一个执行命令的方法
    public abstract void execute();
}
